src/main/java/pl/wrapper/parking/result/Result.java

```diff
@@ -3,6 +3,7 @@
 
 import lombok.SneakyThrows;
 
+
 public interface Result<T> {
```

Comment:
Let's make result have two parameters: Result<someErrorType , T>. Add methods to SuccessResult and FailureResult:
- 'T get()'  that returns the T value and throws is this.isSuccess() = false
- 'someErrorType getError()' that returns someErrorType and throws is this.isSuccess() = true

Reply:
Think of a way to implement a good someErrorType class/record/interface